OpenRA.Mods.Common/Traits/ThrowsParticle.cs

```diff
@@ -66,20 +67,28 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
 
 			// TODO: Carry orientation over from the parent instead of just facing
 			var dynamicFacingInit = init.GetOrDefault<DynamicFacingInit>();
-			var bodyFacing = dynamicFacingInit != null ? dynamicFacingInit.Value() : init.GetValue<FacingInit, int>(0);
-			facing = WAngle.FromFacing(Turreted.TurretFacingFromInit(init, info, 0)());
+			var bodyFacing = dynamicFacingInit != null ? dynamicFacingInit.Value() : init.GetValue<FacingInit, WAngle>(WAngle.Zero);
+			facing = Turreted.TurretFacingFromInit(init, info, WAngle.Zero)();
 
 			// Calculate final position
-			var throwRotation = WRot.FromFacing(Game.CosmeticRandom.Next(1024));
+			var throwRotation = WRot.FromYaw(new WAngle(Game.CosmeticRandom.Next(1024)));
 			var throwDistance = Game.CosmeticRandom.Next(info.MinThrowRange.Length, info.MaxThrowRange.Length);
 
-			initialPos = pos = info.Offset.Rotate(body.QuantizeOrientation(self, WRot.FromFacing(bodyFacing)));
+			initialPos = pos = info.Offset.Rotate(body.QuantizeOrientation(self, WRot.FromYaw(bodyFacing)));
 			finalPos = initialPos + new WVec(throwDistance, 0, 0).Rotate(throwRotation);
 			angle = new WAngle(Game.CosmeticRandom.Next(info.MinThrowAngle.Angle, info.MaxThrowAngle.Angle));
 			length = (finalPos - initialPos).Length / info.Velocity;
 
 			// Facing rotation
-			rotation = WAngle.FromFacing(WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024);
+			var rotationSpeed = WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024;
+			direction = 1;
+			if (rotationSpeed < 0)
+			{
+				rotation = -rotation;
```

Comment:
I don't get this here: Why do you invert `rotation` here if it is getting reassign below regardless? (Mabye wants to be `rotationSpeed`?)